slave/build.py

```diff
@@ -243,6 +243,34 @@ def objdir(self):
     def binary(self):
         return os.path.join(self.objdir(), 'd8')
 
+class ServoBuilder(Builder):
+    def __init__(self, config, folder):
+        super(ServoBuilder, self).__init__(config, folder);
+        # Some other config here
+    
+    def retrieveInfo(self):
+        info = {}
+        info["engine_type"] = "servo"
+        return info
+
+    def objdir(self):
+        return os.path.join(self.folder, 'target')
+
+    def binary(self):
+        return os.path.join(self.objdir(), 'release', 'servo')
+
+    def make(self):
+        # Remember cwd
+        cwd = os.getcwd()
+        # cd into servo's root directory. Requirement of 'mach build'
+        os.chdir(self.folder)
```

Comment:
Use
with utils.FolderChanger(self.folder):

This contains the logic to cd into the folder and when going out of the scope/block it will go back to original cwd